packages/mint-components/src/components/views/copy-text-view.tsx

```diff
@@ -0,0 +1,150 @@
+import { Host, h } from "@stencil/core";
+import { createStyleSheet } from "../../styling/JSS";
+import { HostBlock, P } from "../../global/mixins";
+
+export interface CopyTextViewProps {
+  copyString: string;
+  tooltiptext: string;
+  open: boolean;
+  copyButtonLabel?: string;
+  disabled?: boolean;
+  textAlign?: "left" | "center" | "right";
+  buttonStyle?: "button outside" | "button below" | "icon";
+  error?: boolean;
+  errorText?: string;
+  inputPlaceholderText?: string;
+  dateAvailable?: string;
+  loading?: boolean;
+
+  onClick?: () => void;
+}
+
+const style = {
+  HostBlock: HostBlock,
+  inputStyle: {
+    "&::part(input)": { textOverflow: "ellipsis" },
+    "&::part(base)": { cursor: "pointer" },
+    width: "100%",
+  },
+  inputErrorStyle: {
+    "&::part(base)": {
+      border: "2px solid red",
+    },
+  },
+  containerStyle: {
+    display: "flex",
+    alignItems: "center",
+    gap: "8px",
+  },
+  errorTextStyle: {
+    margin: "0",
+    color: "red",
```

Comment:
Could use a shoelace color CSS variable (the registration/login form should have the default error one we use)